File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string x = "";
        for(int i=0; i<s.size(); i++) {
            if(isalnum(s[i])) x += tolower(s[i]);
        }
        int low = 0, high = x.size() - 1;
        while(low < high) {
            if(x[low] != x[high]) return false;
            low++,high--;
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String str = "";
        for(char ch : s.toCharArray()) {
            if(Character.isDigit(ch)) str += ch;
            if(Character.isLetter(ch)) str += Character.toLowerCase(ch);
        }
        int n = str.length();
        int left = 0, right = n-1;
        while(left < right) {
            if(str.charAt(left) != str.charAt(right)) return false;
            left++;
            right--;
        }
        return true;
    }
}